Error calculating max stack value

```
09:05:56.127   1:7>/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018: The "PublicizeAssemblies" task failed unexpectedly. [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018: dnlib.DotNet.Writer.ModuleWriterException: Error calculating max stack value. If the method's obfuscated, set CilBody.KeepOldMaxStack or MetadataOptions.Flags (KeepOldMaxStack, global option) to ignore this error. Otherwise fix your generated CIL code so it conforms to the ECMA standard. Error occurred after metadata event BeginWriteMethodBodies during writing method 'System.Void DV.DopplerEffects.DopplerFixedUpdateSystem::OnUpdate()' (0x060055EA). [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at dnlib.DotNet.DummyLogger.Log(Object sender, LoggerEvent loggerEvent, String format, Object[] args) [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at dnlib.DotNet.Writer.ModuleWriterBase.dnlib.DotNet.ILogger.Log(Object sender, LoggerEvent loggerEvent, String format, Object[] args) [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at dnlib.DotNet.Writer.Metadata.Error(String message, Object[] args) [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at dnlib.DotNet.Writer.Metadata.dnlib.DotNet.Writer.IWriterError.Error(String message) [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at dnlib.DotNet.Writer.MethodBodyWriter.ErrorImpl(String message) [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at dnlib.DotNet.Writer.MethodBodyWriterBase.GetMaxStack() [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at dnlib.DotNet.Writer.MethodBodyWriter.Write() [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at dnlib.DotNet.Writer.Metadata.WriteMethodBodies() [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at dnlib.DotNet.Writer.Metadata.Create() [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at dnlib.DotNet.Writer.Metadata.CreateTables() [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at dnlib.DotNet.Writer.ModuleWriter.WriteImpl() [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at dnlib.DotNet.Writer.ModuleWriterBase.Write(Stream dest) [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at Publicizer.PublicizeAssemblies.Execute() [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
/home/orion/.nuget/packages/krafs.publicizer/2.0.0/build/Krafs.Publicizer.targets(9,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/home/orion/dv-mods/r93/dv-sandbox/Sandbox.csproj]
                   Done executing task "PublicizeAssemblies" -- FAILED. (TaskId:49)
```

Since Publicizer doesn't modify method bodies but only their access flags, it should be safe to KeepOldMaxStack, as described here: https://stackoverflow.com/questions/58926381/setting-cilbody-keepoldmaxstack-or-metadataoptions-flags

--- a/src/Publicizer/PublicizeAssemblies.cs
+++ b/src/Publicizer/PublicizeAssemblies.cs
@@ -2,8 +2,10 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Security.Cryptography;
 using System.Text;
 using dnlib.DotNet;
+using dnlib.DotNet.Writer;
 using Microsoft.Build.Framework;
 using Microsoft.Build.Utilities;
 
@@ -74,7 +76,14 @@ public class PublicizeAssemblies : Task
                 }
 
                 using var fileStream = new FileStream(outputAssemblyPath, FileMode.Create, FileAccess.ReadWrite, FileShare.Read);
-                module.Write(fileStream);
+
+                var writerOptions = new ModuleWriterOptions(module);
+
+                // Writing the module sometime fails without this flag due to how it was originally compiled.
+                // https://github.com/krafs/Publicizer/issues/42
+                writerOptions.MetadataOptions.Flags |= MetadataFlags.KeepOldMaxStack;
+
+                module.Write(fileStream, writerOptions);
             }
             referencePathsToDelete.Add(reference);
             ITaskItem newReference = new TaskItem(outputAssemblyPath);

--- a/src/Publicizer/Publicizer.csproj
+++ b/src/Publicizer/Publicizer.csproj
@@ -27,7 +27,7 @@
     <PackageTags>msbuild accesschecks public publicizer</PackageTags>
     <PackageIcon>icon.png</PackageIcon>
     <PackageReadmeFile>README.md</PackageReadmeFile>
-    <Version>2.0.0</Version>
+    <Version>2.0.1-beta</Version>
   </PropertyGroup>
 
   <ItemGroup>

--- a/src/Publicizer/Publicizer.csproj
+++ b/src/Publicizer/Publicizer.csproj
@@ -27,7 +27,7 @@
     <PackageTags>msbuild accesschecks public publicizer</PackageTags>
     <PackageIcon>icon.png</PackageIcon>
     <PackageReadmeFile>README.md</PackageReadmeFile>
-    <Version>2.0.1-beta</Version>
+    <Version>2.0.1</Version>
   </PropertyGroup>
 
   <ItemGroup>

